autoload/go/guru.vim

```diff
@@ -198,23 +197,26 @@ function! s:async_guru(args) abort
   endfunction
 
   let start_options = {
-        \ 'close_cb': function("s:close_cb"),
+        \ 'close_cb': funcref("s:close_cb"),
+        \ 'in_mode': 'raw',
+        \ 'out_mode': 'nl',
+        \ 'err_mode': 'nl',
         \ }
 
-  if &modified
-    let l:tmpname = tempname()
-    call writefile(split(result.stdin_content, "\n"), l:tmpname, "b")
-    let l:start_options.in_io = "file"
-    let l:start_options.in_name = l:tmpname
-  endif
-
```

Comment:
There won't be a race condition with guru using this method. The `-modified` flag tells guru to read archives from stdin, and it reads from stdin until EOF before doing anything else.

With that in mind, I can revert that portion of this PR if you still want me to, but I'd rather see it left in due to the performance advantage and elimination of the need to manage temp files (note that the temporary file wasn't being deleted either)...